dts/bindings/usb/uvc/zephyr,uvc-control.yaml

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2024 tinyVision.ai Inc.
```

Comment:
What device does it describe? At first glance, I could not find a device behind it, as there is one in the HID driver.

Reply:
It is a small USB device tree encoded in the Zephyr devicetree, using the USB semantics.

This same concept seems present in UAC:
https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/usb/uac2/zephyr%2Cuac2-input-terminal.yaml#L8
![scrot_20241011_110023_774x715](https://github.com/user-attachments/assets/7cd96303-ecc8-4db7-ae41-cc59b1101e48)


The UVC device need to describe their internal video pipeline to the host so it can address which element it wants to control, and learn which controls are available for every part. There is not a 1 to 1 match with the actual devices

For instance, image sensors always have exposure and gain controls built-in. Exposure is under the Camera Terminal category. Gain is under the Processing Unit category. Each image sensor needs 2 devices to expose all their feature... Or if including a test pattern, 3 (adding the Selector Unit to select the test pattern), or if including JPEG encoding (like OV5640), 4 (adding the Encoding Unit).

I will add a small diagram to explain this, even if this approach is not kept, this will help for the review.

Reply:
Sorry I think I misunderstood your question earlier. You probably meant which `DEVICE_DT_DEFINE(node...)` is defined for all these extra controls, and that is true: there is none.

I will try to reuse the graph of the actual video devices (i.e. ov5640 device, DMCI device...) to generate the descriptors without the user having to worry about it.

Worth trying as then here would not be any configuration needed for the controls, but generic APIs to declare, for each device, which controls are to be exposed from the drivers.

Reply:
I confirm it is possible to scan the devicetree to generate this kind of "video device graph" that represents 1;1 the Zephyr graph. If the format descriptor can further be generated at runtime, there would be no more configuration needed, and therefore no more `compatible =` except `zephyr,uvc`:

```hcl
&zephyr_udc0 {
        uvc: uvc {
                compatible = "zephyr,uvc";
                port {
                        uvc_in: endpoint {
                                direction = "in";
                                remote-endpoint-label = "mipi0_out";
                        };
                };
        };
};
```

Reply:
The latest force-push shows it is possible to use this format provided an extra `formats = <...>` entry is given: Only one `compatible=` for UVC is used in total.